opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -994,4 +994,232 @@ mod tests {
+        async fn export(&mut self, batch: LogBatch<'_>) -> LogResult<()> {
+            // Simulate minimal dependency on tokio by sleeping for a short duration
+            sleep(Duration::from_millis(50)).await;
+
+            let logs = Arc::clone(&self.logs);
+            let mut logs_lock = logs.lock().unwrap();
+            for (log_record, instrumentation) in batch.iter() {
+                logs_lock.push((log_record.clone(), instrumentation.clone()));
+            }
+            Ok(())
+        }
+    }
+
+    #[test]
+    fn test_simple_processor_async_exporter_without_runtime() {
+        // Use `catch_unwind` to catch the panic caused by missing Tokio runtime
+        let result = std::panic::catch_unwind(|| {
+            let exporter = LogExporterThatRequiresTokioSpawn::new();
+            let processor = SimpleLogProcessor::new(Box::new(exporter.clone()));
+
+            let mut record: LogRecord = Default::default();
+            let instrumentation: InstrumentationLibrary = Default::default();
+
+            // This will panic because `tokio::spawn` or an async operation is called without a runtime.
+            processor.emit(&mut record, &instrumentation);
+        });
+
+        // Verify that the panic occurred and check the panic message for the absence of a Tokio runtime
+        assert!(
+            result.is_err(),
+            "The test should fail due to missing Tokio runtime, but it did not."
+        );
+
+        if let Err(panic) = result {
+            let panic_message = panic
+                .downcast_ref::<String>()
+                .map(|s| s.as_str())
+                .or_else(|| panic.downcast_ref::<&str>().copied())
+                .unwrap_or("No panic message");
+
+            assert!(
+                panic_message.contains("no reactor running")
+                    || panic_message
+                        .contains("must be called from the context of a Tokio 1.x runtime"),
+                "Expected panic message about missing Tokio runtime, but got: {}",
+                panic_message
+            );
+        }
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+    async fn test_simple_processor_async_exporter_with_runtime() {
+        let exporter = LogExporterThatRequiresTokioSpawn::new();
+        let processor = SimpleLogProcessor::new(Box::new(exporter.clone()));
+
+        let mut record: LogRecord = Default::default();
+        let instrumentation: InstrumentationLibrary = Default::default();
+
+        processor.emit(&mut record, &instrumentation);
+
+        assert_eq!(exporter.len().await, 1);
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+    #[ignore]
+    // all threads are blocked running blocked::export(), and the exporter further needs tokio
+    // runtime to progress on this blocked thread, resulting in deadlock.
+    async fn test_simple_processor_async_exporter_with_multi_thread_runtime_all_cores_blocked() {
```

Comment:
The issue is due to worker threads getting blocked not the cores.

```suggestion
    async fn test_simple_processor_async_exporter_with_all_runtime_worker_threads_blocked() {
```